test/unit/spec/nbextensions/bulkImportCell/bulkImportCell-spec.js

```diff
@@ -846,5 +846,49 @@ define([
                 });
             });
         });
+
+        const iconTestCases = {
+            editingComplete: true,
+            editingIncomplete: true,
+            launching: false,
+            inProgress: false,
+            inProgressResultsAvailable: false,
+            jobsFinishedResultsAvailable: false,
+            jobsFinished: false,
+            error: false,
+        };
+        for (const [state, expectEval] of Object.entries(iconTestCases)) {
+            it(`Should ${
+                iconTestCases[state] ? '' : 'not '
+            }evaluate params when starting in ${state} state`, async () => {
+                // if the cell is starting in a state where we don't expect params to
+                // be evaluated, then the call to initCell will set up these spies.
+                if (!expectEval) {
```

Comment:
There's a touch of trickiness to work with the `initCell` function. That uses spies in certain cases, which are the same spies we want to listen to! Just make sure that they don't get set twice.